src/responder/common/cache_req/cache_req_search.c

```diff
@@ -268,6 +269,41 @@ static errno_t cache_req_search_dp(struct tevent_req *req,
 static void cache_req_search_oob_done(struct tevent_req *subreq);
 static void cache_req_search_done(struct tevent_req *subreq);
 
+static bool
+cache_req_should_be_in_cache (struct cache_req *cr,
+                              struct ldb_result *result)
+{
+    unsigned int id = 0;
+    const char *object_class;
+
+
+    if (result == NULL || result->count == 0) {
+        /* can't decide so keep it */
+        return true;
+    }
+
+    object_class = ldb_msg_find_attr_as_string(result->msgs[0],
+                                               SYSDB_OBJECTCATEGORY, NULL);
+    if (! object_class) {
+        /* no object_class => can't decide so keep it in cache */
+        return true;
+    }
+
+    if (strcasecmp(object_class, SYSDB_USER_CLASS) == 0) {
+        id = ldb_msg_find_attr_as_uint(result->msgs[0], SYSDB_UIDNUM, 0);
+    } else if (strcasecmp(object_class, SYSDB_GROUP_CLASS) == 0) {
+        id = ldb_msg_find_attr_as_uint(result->msgs[0], SYSDB_GIDNUM, 0);
+    }
```

Comment:
Looking at sdap_save_user() we check whether *both* UID and GID are within the domain limits. Therefore wouldn't it be more consistent to always check the ID if present (then, UID won't be present for groups and the check will be skipped..)

You can also use the `OUT_OF_ID_RANGE` helper macro.

Reply:
Good idea, thanks. PR updated to be consistent with sdap_user_save ().